appstore/validator_test.go

```diff
@@ -161,13 +162,163 @@ func TestVerify(t *testing.T) {
 	if err == nil {
 		t.Errorf("error should be occurred because of timeout")
 	}
+}
+
+func TestVerifyBadURL(t *testing.T) {
+	client := New()
+	client.ProductionURL = "127.0.0.1"
```

Comment:
In other words, I do not see much point being able to inject URLs for testing

Reply:
I have just read your old comment... it sounds like 100% testing coverage is the motivation?

Reply:
The `return err`s won't be covered as well as https://github.com/dogenzaka/go-iap/pull/45/files#diff-952f4b7c6719d9d57683f6d309e8f2cfR136 block of code since it is dependent on the server returning a 21007. Its those pesky `if err != nil`s that are bringing the test coverage down 😭

Reply:
I see... what about extracting methods and mocking them? This way you can test certain case effectively. Like, you could define RequestSandbox method and mock it. Perhaps mocking will simplify the tests? dunno. I am fine with your way too, as they are optional fields anyways.